### PR TITLE
WIP: Add a sidecar to Galley for the current installer

### PR DESCRIPTION
**Pull request by ozevren**

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/galley/values.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 enabled: true
 replicaCount: 1
 image: galley
+sidecar: true
```
Comment (howardjohn) [inline install/kubernetes/helm/istio/charts/galley/values.yaml]: will this still work if sidecar is false?

File: pilot/docker/envoy_policy.yaml.tmpl [hunk 2]
```diff
@@ -73,6 +73,32 @@ static_resources:
 {{- end }}
     type: STRICT_DNS
     dns_lookup_family: "{{ .dns_lookup_family }}"
+  - name: out.galley.15019
+    http2_protocol_options: {}
+    connect_timeout: 1.000s
+    type: STRICT_DNS
```
Comment (howardjohn) [inline pilot/docker/envoy_policy.yaml.tmpl]: Looks like this is missing some fields the others have, at least `  dns_lookup_family: "{{ .dns_lookup_family }}"`?

File: install/kubernetes/helm/istio/charts/galley/values.yaml (hunk 1 above)
Comment (ozevren) [reply install/kubernetes/helm/istio/charts/galley/values.yaml]: No, given that this is a WIP PR. The port # changes, along with the mTLS settings. Both need to be accounted for, which this PR doesn't do yet.

Comment (howardjohn) [reply install/kubernetes/helm/istio/charts/galley/values.yaml]: sgtm, was just pointing it out in case it helped, I saw it was a WIP though

Comment (ozevren) [reply install/kubernetes/helm/istio/charts/galley/values.yaml]: Sorry, didn't mean to be curt. It took me a while to get the whole templating thing just right to get it ported from the installer and merging with the istio API idioms. This is just to get the hello world up and running. I haven't done the full gamut of "what can go wrong here" walkthrough yet. I am sure more things will come up.

**Pull request closed without merging**